### PR TITLE
Pass boolean properties to Redpanda configuration

### LINKED ISSUE #470: Cannot set boolean cluster config settings false
### What happened?

`{{- if $value }}` in https://github.com/redpanda-data/helm-charts/blob/main/charts/redpanda/templates/post-upgrade.yaml#L68
`{{- if $element }}` in https://github.com/redpanda-data/helm-charts/blob/main/charts/redpanda/templates/configmap.yaml#L74 and https://github.com/redpanda-data/helm-charts/blob/main/charts/redpanda/templates/configmap.yaml#L118

Because if I am reading these correctly # enable_idempotence: false                                    # Enable idempotent producer in https://github.com/redpanda-data/helm-charts/blob/main/charts/redpanda/values.yaml#L744
would never be set and the default value for enable_idempotence is true https://github.com/redpanda-data/redpanda/blob/dev/src/v/config/configuration.cc#L560-L565

Don't allow setting any values to boolean `false`. 

### What did you expect to happen?

Allow key/values where the value is `typeOf` boolean.

### How can we reproduce it (as minimally and precisely as possible)?. Please include values file.

.

### Anything else we need to know?

_No response_

### Which are the affected charts?

Redpanda

### Chart Version(s)

.

### Cloud provider

.

### PR DESCRIPTION
**Pull request by RafalKorepta**
fixes #470

### CODE REVIEW

**Comment by RafalKorepta** [comment]
Screenshot 2023-04-28 at 12 15 08

https://github.com/redpanda-data/helm-charts/actions/runs/4829528122/jobs/8604675668

Dedicated test that should be run only in origin not in fork is correctly detected by Github Action and it is cancelled.

File: charts/redpanda/templates/configmap.yaml [hunk 1]
```diff
@@ -53,6 +53,37 @@ limitations under the License.
...
+"enable_transactions"
+"enable_usage"
+"features_auto_enable"
+"kafka_enable_authorization"
+"kafka_enable_partition_reassignment"
+"kafka_qdc_enable"
+"legacy_group_offset_retention_enabled"
+"memory_abort_on_alloc_failure"
+"release_cache_on_segment_roll"
+"storage_strict_data_init" -}}
```
Comment (RafalKorepta) [inline charts/redpanda/templates/configmap.yaml:86]: The boolean properties was taken from running cluster v23.1.7 by the following command:

```
kubectl exec redpanda-0 -- curl -sk https://127.0.0.1:9644/v1/cluster_config/schema | jq -c  '.properties | to_entries | .[] | select(.value.type | contains("boolean")) | .key'
```

Comment (RafalKorepta) [reply charts/redpanda/templates/configmap.yaml:86]: Updated as 23.1.8 added new boolean `use_fetch_scheduler_group`

File: charts/redpanda/templates/configmap.yaml [hunk 2]
```diff
@@ -71,7 +107,10 @@ data:
   {{- end }}
   {{- with (dig "cluster" dict .Values.config) }}
       {{- range $key, $element := .}}
-        {{- if $element }}
```
Comment (alejandroEsc) [inline charts/redpanda/templates/configmap.yaml:74]: We could do something like https://helm.sh/docs/chart_template_guide/function_list/#type-functions
how about `{{ if or (eq (typeOf $element) boolean) $element }}` that way we do not have to know an exhaustive list of boolean configs. We leave the responsibility of the user to know what they are doing and we offload maintenance.

Comment (RafalKorepta) [reply charts/redpanda/templates/configmap.yaml:74]: Sure I can change that.

**Comment by joejulian** [comment]
rebased

**Review by joejulian — APPROVED** [review]

**Pull request merged**